shared/source/compiler_interface/linux/os_compiler_cache_helper.cpp

```diff
@@ -45,10 +45,18 @@ bool checkDefaultCacheDirSettings(std::string &cacheDir, NEO::EnvironmentVariabl
     cacheDir = reader.getSetting("XDG_CACHE_HOME", emptyString);
 
     if (cacheDir.empty()) {
+#ifdef ANDROID
+        cacheDir = "/data/data";
+        cacheDir = joinPath(cacheDir, getprogname());
```

Comment:
What's expected behavior for a cmdline binary?  will /data/data/<progname> creation/access fail or will it still work?

Reply:
/data/data/progame I meant

Reply:
I tested with the tflite benchmark, by giving execution permission to benchmark when logged in as a root user, I Didn't see any errors though, in data/data/tflite_benchmark/.cache/neo_compiler_cache  directory is getting created and being used

Reply:
Updated code to provide fallback to /data/local/tmp/cache/progname/.cache/neo_compiler_cache directory, in case process doesn't have permission for /data/data/